### ISSUE TITLE
Consider replacing JMImageCache with SDWebImage?

### ISSUE THREAD

**Issue by markrickert** [clearsightstudio/ProMotion#683]
`JMImageCache` doesn't have any way to cancel the operation.

I'm running into problems where when I've got a LOT of cells with remote images and i start scrolling through the images and then stop and then scroll again, images start appearing that are wrong and then the right ones overwrite them. This is happening because of the cell reuse and I need to be able to cancel the image request when the `prepareForReuse` method is invoked.

I've taken it upon myself to do this with `SDWebImage` in a custom cell subclass:

```ruby
def table_data
  [{
    title: nil,
    cells: [{
      title: name,
      cell_class: MyRemoteImageCell,
      properties: {
        remote_image: "http://whatever/",
        placeholder_image: "local_placeholder_image"
      }      
    }]
  }]
end

class MyRemoteImageCell < PM::TableViewCell

  def initWithStyle(style_name, reuseIdentifier: reuseIdentifier)
    super
    self
  end

  def prepareForReuse
    @image_operation.cancel if @image_operation
    @image_operation = nil
  end

  def remote_image=(url)
    @image_operation = manager.downloadWithURL(NSURL.URLWithString(url),
      options:SDWebImageRefreshCached,
      progress:nil,
      completed: -> image, error, cacheType, finished {
        self.imageView.image = image unless image.nil?
    })
  end

  def placeholder_image=(image)
    self.imageView.image = UIImage.imageNamed(image)
  end

  def manager
    SDWebImageManager.sharedManager
  end

end
```

**Comment by jamonholmgren** [comment]
Back to our roots? https://github.com/clearsightstudio/ProMotion/pull/394

:smile:

**Comment by jamonholmgren** [comment]
We could probably support both. Their APIs are so similar, for the most part.

**Comment by markrickert** [comment]
The problem with JMImageCache though is that it can't cancel the operation so it wounds up setting the image of a reused cell when there's a slow internet connection and there's a lot of ell images and scrolling going on.

**Comment by jamonholmgren** [comment]
I'm not opposed to reverting back to SDWebImage.

**Comment by markrickert** [comment]
Here's an example of this bug with an easy to see gif of the problem :)

**Comment by markrickert** [comment]
Closed by #688

**Comment by jamonholmgren** [comment]
:+1:
